src/Controller/HelperController.php

```diff
@@ -64,22 +65,40 @@ class HelperController
     protected $validator;
 
     /**
-     * @param ValidatorInterface $validator
+     * @var DataTransformerResolver
      */
-    public function __construct(Environment $twig, Pool $pool, AdminHelper $helper, $validator)
+    private $resolver;
+
+    /**
+     * @param ValidatorInterface           $validator
+     * @param DataTransformerResolver|null $resolver
+     */
+    public function __construct(Environment $twig, Pool $pool, AdminHelper $helper, $validator, $resolver = null)
     {
         // NEXT_MAJOR: Move ValidatorInterface check to method signature
         if (!($validator instanceof ValidatorInterface)) {
             throw new \InvalidArgumentException(
-                'Argument 4 is an instance of '.\get_class($validator).', expecting an instance of'
-                .' \Symfony\Component\Validator\Validator\ValidatorInterface'
+                'Argument 4 is an instance of '.\get_class($validator).', expecting an instance of '
+                .ValidatorInterface::class
             );
         }
 
+        // NEXT_MAJOR: Move DataTransformerResolver check to method signature
+        if (!($resolver instanceof DataTransformerResolver)) {
+            @trigger_error(sprintf(
+                'Passing other type than %s in argument 4 to %s() is deprecated since sonta-project/admin-bundle 3.x and will throw %s exception in 4.0.',
```

Comment:
```suggestion
                'Passing other type than %s in argument 4 to %s() is deprecated since sonata-project/admin-bundle 3.x and will throw %s exception in 4.0.',
```